### PR TITLE
Remove obsolete APIs for Debug and Profiling in PCIe, Zynq Device and Hw Emu flow

### PR DESCRIPTION
**Pull request by IshitaGhosh**

### CODE REVIEW

File: src/runtime_src/core/edge/user/shim.cpp [hunk 1]
```diff
@@ -1231,6 +1231,17 @@ int ZYNQShim::xclLogMsg(xrtLogMsgLevel level, const char* tag,
     else return 0;
   }
 
+  double ZYNQShim::xclGetDeviceClockFreqMHz()
+  {
+    xclDeviceInfo2  deviceInfo ;
+    xclGetDeviceInfo2(&deviceInfo) ;
+    unsigned clockFreq = deviceInfo.mOCLFrequency[0] ;
+    if (clockFreq == 0) clockFreq = 100 ;
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp]: debugging is tricky when you add condition and expression on same line

File: src/runtime_src/core/edge/user/shim.cpp [hunk 2]
```diff
@@ -1231,6 +1231,17 @@ int ZYNQShim::xclLogMsg(xrtLogMsgLevel level, const char* tag,
     else return 0;
   }
 
+  double ZYNQShim::xclGetDeviceClockFreqMHz()
+  {
+    xclDeviceInfo2  deviceInfo ;
+    xclGetDeviceInfo2(&deviceInfo) ;
+    unsigned clockFreq = deviceInfo.mOCLFrequency[0] ;
```
Comment (stsoe) [inline src/runtime_src/core/edge/user/shim.cpp]: unsigned what?  please fix in all files you have changed.
Ironically, I wouldn't have asked if you had used 'auto' :-)

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx [hunk 3]
```diff
@@ -537,6 +390,19 @@ int xclPollCompletion(xclDeviceHandle handle, int min_compl, int max_compl, xclR
...
 }
 
+size_t xclDebugReadIPStatus(xclDeviceHandle handle, xclDebugReadType type, void* debugResults)
+{
+  return 0;
+}
+
+double xclGetDeviceClockFreqMHz(xclDeviceHandle handle)
+{
+  xclhwemhal2::HwEmShim *drv = xclhwemhal2::HwEmShim::handleCheck(handle);
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx]: auto?

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx [hunk 4]
```diff
@@ -2369,6 +2370,46 @@ int HwEmShim::xclReadTraceData(void* traceBuf, uint32_t traceBufSz, uint32_t num
...
+  if (type == XCL_PERF_MON_MEMORY) {
+    str = (slotnum < XAIM_MAX_NUMBER_SLOTS) ? mPerfMonSlotName[slotnum] : "";
+  }
+  if (type == XCL_PERF_MON_ACCEL) {
+    str = (slotnum < XAM_MAX_NUMBER_SLOTS) ? mAccelMonSlotName[slotnum] : "";
+  }
+  if (type == XCL_PERF_MON_STR) {
+    str = (slotnum < XASM_MAX_NUMBER_SLOTS) ? mStreamMonSlotName[slotnum] : "";
+  }
+  strncpy(slotName, str.c_str(), length);
```
Comment (stsoe) [inline src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/shim.cxx]: null termination is not guaranteed when you copy at most length chars and don't assert that str.length() is less.

File: src/runtime_src/core/pcie/linux/shim.cpp [hunk 5]
```diff
@@ -1527,6 +1522,52 @@ int shim::xclReadTraceData(void* traceBuf, uint32_t traceBufSz, uint32_t numSamp
     return size;
 }
 
+// Get the device clock frequency (in MHz)
+double shim::xclGetDeviceClockFreqMHz()
+{
+  xclGetDeviceInfo2(&mDeviceInfo);
+  unsigned clockFreq = mDeviceInfo.mOCLFrequency[0];
```
Comment (stsoe) [inline src/runtime_src/core/pcie/linux/shim.cpp]: auto or type

File: src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx (hunk 3 above)
Comment (IshitaGhosh) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx]: Is auto required here? The required type is already used here.

Comment (stsoe) [reply src/runtime_src/core/pcie/emulation/hw_em/generic_pcie_hal2/halapi.cxx]: It's not.

**Pull request merged**